### PR TITLE
feat: add tags selection to feature creation

### PR DESCRIPTION
**Pull request by sjaanus**
image

### CODE REVIEW

File: frontend/src/component/feature/hooks/useFeatureForm.ts [hunk 1]
```diff
@@ -48,11 +50,17 @@ const useFeatureForm = (
     }, [initialImpressionData]);
 
     const getTogglePayload = () => {
+        const splitTags: ITag[] = Array.from(tags).map((tag) => {
```
Comment (sjaanus) [inline frontend/src/component/feature/hooks/useFeatureForm.ts:53]: Need to do some extra mapping due to dropdown value being string not object.

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx [hunk 2]
```diff
@@ -262,6 +272,34 @@ export const CreateFeatureDialog = ({
...
                             />
+                            <MultiSelectConfigButton
+                                tooltip={{
+                                    header: 'Select tags',
+                                }}
+                                description={configButtonData.tags.text}
+                                selectedOptions={tags}
+                                options={allTags.map((tag) => ({
+                                    label: `${tag.type}:${tag.value}`,
+                                    value: `${tag.type}:${tag.value}`,
```
[the remark this answers is not in the document]
Comment (thomasheartman) [reply frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx:283]: Yeah, I ran into the same issue for the impression data one. I agree with your assessment here, but I'd like to try and have another look myself when this is in.

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx [hunk 3]
```diff
@@ -262,6 +272,34 @@ export const CreateFeatureDialog = ({
...
+                                }))}
+                                onChange={setTags}
+                                button={{
+                                    label:
+                                        tags.size > 0
+                                            ? `${tags.size} selected`
+                                            : 'Tags',
+                                    labelWidth: `9ch`,
```
Comment (thomasheartman) [inline frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx:291]: What's the label width based on here? Maybe we could do something like:
```suggestion
                                    labelWidth: `${"nn selected".length}ch`,
```

to make it clear what it needs to fit? I also think it should be at least 10ch long, but probably 11 😅

**Review by thomasheartman — APPROVED** [review]

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx (hunk 3 above)
Comment (sjaanus) [reply frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx]: Makes sense. Updated

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomasheartman) [reply frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx:283]: Think I got it working, actually. I'll have a look when you've merged this

[the remark this answers is not in the document]
Comment (thomasheartman) [reply frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureTogglesHeader/CreateFeatureDialog.tsx:283]: Or you could steal the implementation in https://github.com/Unleash/unleash/pull/7676 and try for yourself if you wanna?

**Pull request merged**